pyHalo/Halos/HaloModels/TNFWFromParams.py

```diff
@@ -0,0 +1,146 @@
+from pyHalo.Halos.halo_base import Halo
+from lenstronomy.LensModel.Profiles.tnfw import TNFW as TNFWLenstronomy
+import numpy as np
+from pyHalo.Halos.tnfw_halo_util import tnfw_mass_fraction
+from lenstronomy.LensModel.Profiles.tnfw import TNFW
+
+class TNFWFromParams(Halo):
+
+
+    KEY_RT = "r_trunc_kpc"
+    KEY_RS = "rs"
+    KEY_RHO_S = "rho_s"
+    KEY_RV = "rv"
+
+    mass_bound_uss_massfraction = True 
+
+    """
+    The base class for a truncated NFW halo
+    """
+    def __init__(self, mass, x_kpc, y_kpc, r3d, z,
+                 sub_flag, lens_cosmo_instance, args, unique_tag=None):
+        """
+        Denfines a TNFW subhalo with physical params r_trunc_kpc, rs, rhos passed in the args argument
+        """
+
+        self._lens_cosmo = lens_cosmo_instance
+
+        self._kpc_per_arcsec_at_z = self._lens_cosmo.cosmo.kpc_proper_per_asec(z)
+
+        x = x_kpc / self._kpc_per_arcsec_at_z
+
+        y = y_kpc / self._kpc_per_arcsec_at_z
+
+        self._params_physical = args
```

Comment:
In general args might have other parameters besides the physical parameters describing the NFW profile. Can you extract only the parameters you need from args to define self._params_physical?